test/e2e/plan_test.go

```diff
@@ -0,0 +1,109 @@
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCli_Plans(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping test in short mode.")
+	}
+	subCommand := "plans"
+	consumerToken := ""
+	apiURL := ""
+	Version := "metal"
+	rootClient := root.NewClient(consumerToken, apiURL, Version)
+	type fields struct {
+		MainCmd  *cobra.Command
+		Outputer outputPkg.Outputer
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    *cobra.Command
+		cmdFunc func(*testing.T, *cobra.Command)
+	}{
+		{
+			name: "get",
+			fields: fields{
+				MainCmd:  plans.NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				root := c.Root()
+				out := &bytes.Buffer{}
+				root.SetArgs([]string{subCommand, "get"})
+				root.SetOutput(out)
+				err := root.Execute()
+				assert.NoError(t, err)
+				outStr := out.String()
+				assert.Contains(t, outStr, "m3.small.x86")
+				assert.Contains(t, outStr, "m3.large.x86")
+				assert.Contains(t, outStr, "c3.medium.x86")
+				assert.Contains(t, outStr, "c3.small.x86")
+				assert.Contains(t, outStr, "x2.xlarge.x86")
+				assert.Contains(t, outStr, "x3.xlarge.x86")
+			},
+		},
+		{
+			name: "get",
+			fields: fields{
+				MainCmd:  plans.NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				root := c.Root()
+				out := &bytes.Buffer{}
+				root.SetArgs([]string{subCommand, "get", "--filter ", "slug=m3.small.x86"})
+				root.SetOutput(out)
+				if err := root.Execute(); err != nil {
+					t.Error(err)
+				}
+				if !strings.Contains(out.String(), "m3.small.x86") {
+					t.Error("expected output should include m3.small.x86 by SLUG")
+				}
+			},
+		},
+		{
+			name: "get",
+			fields: fields{
+				MainCmd:  plans.NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				root := c.Root()
+				out := &bytes.Buffer{}
+				root.SetArgs([]string{subCommand, "get", "--filter ", "type=standard"})
+				root.SetOutput(out)
+				if err := root.Execute(); err != nil {
+					t.Error(err)
+				}
+				if !strings.Contains(out.String(), "m3.small.x86") &&
+					!strings.Contains(out.String(), "m3.large.x86") &&
+					!strings.Contains(out.String(), "c3.medium.x86") &&
+					!strings.Contains(out.String(), "c3.small.x86") {
+					t.Error("expected output should include m3.small.x86, m3.large.x86, c3.medium.x86 and c3.small.x86 by SLUG")
+				}
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			rootCmd := rootClient.NewCommand()
+			rootCmd.AddCommand(tt.fields.MainCmd)
```

Comment:
One thing you could try is getting rid of `tt.fields.MainCmd`, and instead doing:

```go
out := &bytes.Buffer{}
testCommand := NewClient(rootClient, outputPkg.Outputer(out)).NewCommand()
rootCmd.AddCommand(testCommand)
tt.cmdFunc(t, rootCmd, out)
```
I haven't tried that, though, so I don't know if `outputPkg.Outputer` will accept a byte buffer.